File: Software/BrickPi_C/Sensor_Examples/LEGO - Motor Test_Timeout.c
/*
*  Karan Nayna
*  karan25991<at>gmail.com
*  Initial date: July 10, 2013
*  Updated:  Feb 17, 2015 (John)
*  Based on Matthew Richardson's Example for testing BrickPi
*  You may use this code as you wish, provided you give credit where it's due.
*
*  This is a program for testing the RPi BrickPi driver with Lego Motors on PortA and Port B with the timeout enabled
*/

//Build Instuction:
//gcc -o program "LEGO - Motor Test.c" -lrt -lm -L/usr/local/lib -lwiringPi -I//home/pi/Desktop/git/BrickPi_C/Drivers #"Path to the Driver Files BrickPi.h and tick.h"
//./program

#include <stdio.h>
#include <math.h>
#include <time.h>
#include "tick.h"
#include "BrickPi.h"
#include <linux/i2c-dev.h>  
#include <fcntl.h>

// Compile Using:
// sudo gcc -o program "LEGO - Motor Test_Timeout.c" -lrt -lm
// Run the compiled program using:
// sudo ./program

int result,v,f;
#undef DEBUG

int main() {
  ClearTick();
  
  result = BrickPiSetup();
  printf("BrickPiSetup: %d\n", result);
  if(result)
    return 0;

  BrickPi.Address[0] = 1;
  BrickPi.Address[1] = 2;

  BrickPi.MotorEnable[PORT_A] = 1;
  BrickPi.MotorEnable[PORT_B] = 1;
  result = BrickPiSetupSensors();
  printf("BrickPiSetupSensors: %d\n", result); 
  v=0;
  f=1;

  //Communication timeout in ms (how long since last valid communication before floating the motors). 
  //0 disables the timeout so the motor would keep running even if it is not communicating with the RaspberryPi
  BrickPi.Timeout=1000;

  printf("BrickPiSetTimeout Status:%d\n",BrickPiSetTimeout()); //BrickPiSetTimeout() returns 0 if timeout has been set 
  if(!result){
    
    usleep(100000);
    
    while(1){
      result = BrickPiUpdateValues();
      if(!result)
       {
			BrickPi.MotorSpeed[PORT_A]=200;
			BrickPi.MotorSpeed[PORT_B]=200;
			BrickPiUpdateValues();		
       }
	
      usleep(100000);
    }
	
  }
  return 0;
}

File: Software/BrickPi_Python/Sensor_Examples/LEGO-Motor_Test_Timeout.py
#!/usr/bin/env python
# Karan Nayan
# Initial Date: July 10, 2013
# Updated:      July 10, 2013
# Updated:      Oct  28, 2016 Shoban
#
# These files have been made available online through a Creative Commons Attribution-ShareAlike 3.0  license.
# (http://creativecommons.org/licenses/by-sa/3.0/)
#
# This code is for testing the BrickPi with a Lego Motor and with the timeout enabled
# Connect the LEGO Motors to Motor ports MA and MB.
#
# You can learn more about BrickPi here:  http://www.dexterindustries.com/BrickPi
# Have a question about this example?  Ask on the forums here:  http://forum.dexterindustries.com/c/brickpi

# For the code to work - sudo pip install -U future

from __future__ import print_function
from __future__ import division
from builtins import input

# the above lines are meant for Python3 compatibility.
# they force the use of Python3 functionality for print(), 
# the integer division and input()
# mind your parentheses!


from BrickPi import *   #import BrickPi.py file to use BrickPi operations

BrickPiSetup()  # setup the serial port for communication

BrickPi.MotorEnable[PORT_A] = 1 #Enable the Motor A
BrickPi.MotorEnable[PORT_B] = 1 #Enable the Motor B

BrickPiSetupSensors()   #Send the properties of sensors to BrickPi

#Communication timeout in ms (how long since last valid communication before floating the motors).
#0 disables the timeout so the motor would keep running even if it is not communicating with the RaspberryPi
BrickPi.Timeout=3000
print("BrickPiSetTimeout Status :",BrickPiSetTimeout())

while True:
    print("Running Forward")
    BrickPi.MotorSpeed[PORT_A] = 200  #Set the speed of MotorA (-255 to 255)
    BrickPi.MotorSpeed[PORT_B] = 200  #Set the speed of MotorB (-255 to 255)
    ot = time.time()
    while(time.time() - ot < 3):    #running while loop for 3 seconds
        BrickPiUpdateValues()       # Ask BrickPi to update values for sensors/motors
        time.sleep(.1)              # sleep for 100 ms
    print("Running Reverse")
    BrickPi.MotorSpeed[PORT_A] = -200  #Set the speed of MotorA (-255 to 255)
    BrickPi.MotorSpeed[PORT_B] = -200  #Set the speed of MotorB (-255 to 255)
    ot = time.time()
    while(time.time() - ot < 3):    #running while loop for 3 seconds
        BrickPiUpdateValues()       # Ask BrickPi to update values for sensors/motors
        time.sleep(.1)              # sleep for 100 ms
